packages/docusaurus-plugin-content-docs/src/index.ts

```diff
@@ -157,6 +157,12 @@ export default async function pluginContentDocs(
         );
 
         const [drafts, docs] = _.partition(docsBase, (doc) => doc.draft);
+        const unlistedIds = docs.reduce<string[]>((acc, doc) => {
```

Comment:
This attribute is only used in a single function

I suggest creating a Set of unlistedIds deeper inside `addDocNavigation`